### PR TITLE
feat: add rust code for build_in_hash

### PR DESCRIPTION
**Pull request by WSL0809**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/rust/chapter_hashing/hash_map_chaining.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        for bucket in &self.buckets {
+            let mut res = Vec::new();
+            for pair in bucket {
+                res.push(format!("{} -> {}", pair.key, pair.val));
+            }
+            println!("{:?}", res);
+        }
+    }
+
+    fn put(&mut self, key: i32, val: String) {
```
Comment (krahets) [inline codes/rust/chapter_hashing/hash_map_chaining.rs:87]: method comment

File: codes/rust/chapter_hashing/hash_map_chaining.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    fn hash_func(&self, key: i32) -> usize {
+        key as usize % self.capacity as usize
+    }
+
+    fn load_factor(&self) -> f32 {
+        self.size as f32 / self.capacity as f32
+    }
+
+    // 删除操作
```
Comment (krahets) [inline codes/rust/chapter_hashing/hash_map_chaining.rs:41]: /* 删除操作 */

**Comment by krahets** [comment]
Hi @WSL0809 , please check the code and comment to make them consistent with the reference code (Java or Python)

**Comment by WSL0809** [comment]
Thanks for the correction, I've changed the function comment @krahets

**Comment by krahets** [comment]
@sjinzh Please review this PR, thx!

File: codes/rust/chapter_hashing/hash_map_chaining.rs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+        for pair in bucket {
+            if pair.key == key {
+                pair.val = val.clone();
+                return;
+            }
+        }
+        let bucket = &mut self.buckets[index];
+
+        // 若无该 key ,则添加新的键值对
```
Comment (sjinzh) [inline codes/rust/chapter_hashing/hash_map_chaining.rs:109]: // 若无该 key ，则将键值对添加至尾部

File: codes/rust/chapter_hashing/hash_map_chaining.rs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        // 若未找到 key 则返回 None
+        None
+    }
+}
+
+/* Driver Code */
+pub fn main() {
+    /* 测试代码 */
```
Comment (sjinzh) [inline codes/rust/chapter_hashing/hash_map_chaining.rs:137]: /* 初始化哈希表 */

**Review by sjinzh — CHANGES_REQUESTED** [review]
@krahets 
Request changes .

**Comment by WSL0809** [comment]
> @krahets Request changes .

Changes made. Please re-review. Thanks!

**Comment by krahets** [comment]
Hi @WSL0809 , could you please implement `hash_map_open_addressing.rust` and push to this PR?

**Comment by WSL0809** [comment]
> Hi @WSL0809 , could you please implement `hash_map_open_addressing.rust` and push to this PR? Because we wonder all the codes in hash collision implemented in one PR.

sure！

**Pull request closed without merging**